tests/utils_tests/test_functional.py

```diff
@@ -130,3 +131,11 @@ def test_lazy_equality(self):
 
         self.assertEqual(lazy_a(), lazy_b())
         self.assertNotEqual(lazy_b(), lazy_c())
+
+    def test_lazy_repr(self):
+        """
+        Tests that a proxy of lazy object's repr equals to the object's value
+        """
+        original_object = 'Lazy translation text'
+        lazy_obj = ugettext_lazy(original_object)
+        self.assertEquals(original_object, repr(lazy_obj))
```

Comment:
@carljm @timgraham I did these tests, which I think cover a wider range of possibilities. Should I leave them all? Remove any? Make new? Suggestions welcome :)

    def test_lazy_repr_text(self):
        original_object = 'Lazy translation text'
        lazy_obj = ugettext_lazy(original_object)
        self.assertEquals(repr(original_object), repr(lazy_obj))

    def test_lazy_repr_lambda_int(self):
        original_object = 15
        lazy_method = lazy(lambda t: t, int)
        lazy_obj = lazy_method(original_object)
        self.assertEquals(repr(original_object), repr(lazy_obj))

    def test_lazy_repr_dict_dict(self):
        original_object = {'test': 'dict', 1: 15}
        lazy_method = lazy(lambda t: t, dict)
        lazy_obj = lazy_method(original_object)
        self.assertEquals(repr(original_object), repr(lazy_obj))

    def test_lazy_repr_lambda_bytes(self):
        original_object = 15
        lazy_method = lazy(lambda t: t, bytes)
        lazy_obj = lazy_method(original_object)
        self.assertEquals(repr(bytes(original_object)), repr(lazy_obj))